pkg/apis/registry/v1alpha3/types.go

```diff
@@ -66,6 +69,19 @@ type GarbageCollection struct {
 	TTL metav1.Duration `json:"ttl"`
 }
 
+// Proxy contains settings for a proxy used in the registry cache.
+type Proxy struct {
+	// HTTPProxy is used as HTTP_PROXY env in the StatefulSet.
+	// +optional
+	HTTPProxy *string `json:"httpProxy"`
+	// HTTPSProxy is used as HTTPS_PROXY env in the StatefulSet.
+	// +optional
+	HTTPSProxy *string `json:"httpsProxy"`
+	// NoProxy is used as NO_PROXY env in the StatefulSet.
+	// +optional
+	NoProxy *string `json:"noProxy"`
+}
```

Comment:
I think we need some validation for the new fields. Ensure that the httpProxy field starts with `http://` and the httpsProxy field starts with `https://`. Are you allowed to specify a port? Validation that field is a valid URL? If the protocol is required, should we have 2 fields? We can have a single field and based on the protocol set the appropriate env var. WDYT?

Reply:
> I think we need some validation for the new fields. Ensure that the httpProxy field starts with http:// and the httpsProxy field starts with https://.

You can use `http://IP` also in `httpsProxy`. In my understanding `httpsProxy` is used for all HTTPS connections and  `httpProxy` is used for all HTTP connections. So using only one field would not work. 

> Are you allowed to specify a port?
Yes, like`http://IP:PORT`.

>  Validation that field is a valid URL? 
That sounds like a good idea, I will add this.